extensions/amp-carousel/0.1/test/test-slidescroll.js

```diff
@@ -13,3 +13,217 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+import * as sinon from 'sinon';
+require('../amp-carousel');
+import {createIframePromise} from '../../../../testing/iframe';
+import {toggleExperiment} from '../../../../src/experiments';
+
+describe('SlideScroll', () => {
+  let sandbox;
+
+  beforeEach(() => {
+    toggleExperiment(window, 'amp-slidescroll', true);
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function getAmpSlideScroll() {
+    return createIframePromise().then(iframe => {
+      toggleExperiment(iframe.win, 'amp-slidescroll', true);
+      const imgUrl = 'https://lh3.googleusercontent.com/5rcQ32ml8E5ONp9f9-' +
+          'Rf78IofLb9QjS5_0mqsY1zEFc=w300-h200-no';
+      const slideScrollHtml =
+        `<amp-carousel type='slides' width=400 height=300 id='c1' controls>
```

Comment:
You can control when `buildCallback` is called. It's the constructor that is a problem. It makes the difference since we are trying to minimize the scope of inner HTML for tests.